creusot/src/translation/external.rs

```diff
@@ -104,10 +104,12 @@ pub(crate) fn extract_extern_specs_from_item<'tcx>(
     let mut subst = Vec::new();
     let mut errors = Vec::new();
     for i in 0..outer_subst.len() {
-        match (inner_subst[i + extra_parameters].unpack(), outer_subst[i].unpack()) {
+        match (inner_subst[i + extra_parameters].kind(), outer_subst[i].kind()) {
             (GenericArgKind::Type(t1), GenericArgKind::Type(t2)) => match (t1.kind(), t2.kind()) {
                 (TyKind::Param(param1), TyKind::Param(param2))
-                    if param1.name == param2.name || param1.name.as_str().starts_with("Self") =>
+                    if true // TODO
+                        || param1.name == param2.name
+                        || param1.name.as_str().starts_with("Self") =>
```

Comment:
Last TODO

Reply:
What's the issue here?

(Apart from the fact that `start_with("Self")` is a very bad way of detecting `Self_`.)

Reply:
The type of the fn argument of `is_some_and` changed from `impl FnOnce(T) -> bool` to `impl [const] FnOnce(T) -> bool + [const] Destruct` (and this type is used as the name of its own Param). So it seems that the right fix is to change the `extern_spec` of `is_some_and` to have a matching type. I haven't found the right feature yet (I think it's `const_trait_impl` but ``impl [const] FnOnce(T) -> bool + [const] Destruct` still refuses to parse.

Reply:
Do we use the latest version of syn?

Reply:
Adding `[const]` to arguments implies making the function itself `const`, which conflicts with our encoding of contracts.

```
error[E0493]: destructor of `std::option::Option<T>` cannot be evaluated at compile-time
  --> creusot-contracts/src/std/option.rs:67:17
```

So maybe it's time to not compare parameters by name?

Reply:
So you call `f` in the body of its extern spec? That looks like an option. Perhaps @xldenis could give his opinion here, because I never touched this part of Creusot, and I don't know what are the tradeoffs.